adapters/repos/db/vector/hnsw/neighbor_connections.go

```diff
@@ -13,20 +13,32 @@ package hnsw
 
 import (
 	"context"
+	"math"
 	"time"
 
 	"github.com/pkg/errors"
 	"github.com/weaviate/weaviate/adapters/repos/db/helpers"
 	"github.com/weaviate/weaviate/adapters/repos/db/priorityqueue"
 	"github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/visited"
 )
 
 func (h *hnsw) findAndConnectNeighbors(node *vertex,
 	entryPointID uint64, nodeVec []float32, distancer compressionhelpers.CompressorDistancer, targetLevel, currentMaxLevel int,
 	denyList helpers.AllowList,
 ) error {
 	nfc := newNeighborFinderConnector(h, node, entryPointID, nodeVec, distancer, targetLevel,
-		currentMaxLevel, denyList)
+		currentMaxLevel, denyList, false)
+
+	return nfc.Do()
+}
+
+func (h *hnsw) findAndConnectNeighborsAfterCleanUpTombstonedNodes(node *vertex,
```

Comment:
Are we able to use something a shorter name for this? What about `findAndConnectNeighborsAfterTombstoneCleanup`?

Reply:
What about `reconnectNeighboursOf`. I have also simplified unused parameters.